### ISSUE TITLE
Setfit classifier trained on a static-embeddings model?

### ISSUE THREAD

**Issue by next-game-solutions** [huggingface/setfit#616]
@tomaarsen I would appreciate it if you could comment on whether it's possible to train a `setfit` model based on a static-embeddings model pretrained, for example, using the `model2vec`, `tokenlearn`, or the sentence-transformers based [approach](https://huggingface.co/blog/static-embeddings) that you described earlier this year?

The `model2vec` library does provide [functionality](https://minish.ai/packages/model2vec/training) for training classifiers, but it seems using `model2vec` for that would require a large amount of training data. Hence my question: could one instead combine the benefit (inference speed) of a pre-trained static-embeddings model with the low data requirements offered by `setfit`?

**Comment by next-game-solutions** [comment]
Thank you very much, @tomaarsen! The boilerplate code you provided is what I was after - to understand if a static-embeddings model can in principle be used as a base model to train a `setfit` classifier. The fact that the resultant classifier is not particularly accurate is something to be expected I guess, given the static nature of embeddings. Perhaps, though, with a larger training dataset it could work better.

I do have a decent static-embeddings model trained for the domain I am working in. I am going to through a bunch of training data onto it to train a `setfit` classifier. The possibility of deploying such a classifer on an edge device or, say, as an AWS Lambda, is too attractive not to try.

Thanks again.

**Comment by next-game-solutions** [comment]
Appreciate your further suggestions, Tom. Yes, I've already figured the importance of a higher learning rate than what normally would be used when training/finetuning a non-static model. Your blog post was very helpful in that regard.
